imblearn/under_sampling/_prototype_selection/_edited_nearest_neighbours.py

```diff
@@ -479,15 +498,22 @@ def _fit_resample(self, X, y):
 
         self.sample_indices_ = np.arange(X.shape[0], dtype=int)
 
-        for curr_size_ngh in range(1, self.nn_.n_neighbors):
```

Comment:
If I understand this line of code correctly, the current version of AllKNN will train in the first round a KNN with **1 neighbour**, and it will run only as many times as neighbours the user enters in the n_neighbours params +1.

Is this the desired behaviour? 

I think the first KNN should have the neighbours specified in n_neighbours (default is 3 and not 1), and it should run as many times as the user wants, unless  one of the stopping criteria are met.

The code below should implement the desired behaviour